src/scene/scene.cpp

```diff
@@ -602,6 +603,48 @@ const std::vector<Scene::InterestingTime>& Scene::interestingTimes() const {
     return _interestingTimes;
 }
 
+void Scene::setFromProfile_properties(const Profile& p) {
+    ghoul::lua::LuaState L(ghoul::lua::LuaState::IncludeStandardLibrary::Yes);
+
+    for (const Profile::Property& prop : p.properties) {
+        std::string uriOrRegex = prop.name;
+        std::string groupName;
+        if (doesUriContainGroupTag(uriOrRegex, groupName)) {
+            // Remove group name from start of regex and replace with '*'
+            uriOrRegex = removeGroupNameFromUri(uriOrRegex);
+        }
+        ghoul::lua::push(L, uriOrRegex);
+        ghoul::lua::push(L, 0.0);
+        // Later functions expect the value to be at the last position on the stack
+        property_pushValueFromProfileToLuaState(L, prop.value);
+
+        applyRegularExpression(
+            L,
+            uriOrRegex,
+            allProperties(),
+            0.0,
+            groupName,
+            ghoul::EasingFunction::Linear
+        );
+        //Clear lua state stack
+        lua_settop(L, 0);
+    }
+}
+
+void Scene::property_pushValueFromProfileToLuaState(ghoul::lua::LuaState& L,
+    const std::string& value)
+{
+    if (!luascriptfunctions::convertStringToLuaAndPush_bool(L, value)) {
+        if (!luascriptfunctions::convertStringToLuaAndPush_float(L, value)) {
```

Comment:
Since the functions in the `if` statement are not used anywhere else, I think it migh tbe better to write them out here as it makes them less magic as to what is happening here.
```
if (value == "true") {
    ghoul::lua::push(L, true);
}
else if (value == "false") {
    ghoul::lua::push(L, false);
}
else {
      try {
        float converted = std::stof(s);
        ghoul::lua::push(L, converted);
    }
    catch (...) {
        ghoul::lua::push(L, value == "nil" ? value : "[[" + value + "]]");
    }
}
```

Reply:
Making the new functions allows me to describe what is happening in the function name itself. I think it's clearer than if I put it all together like that.